Coupled-Cluster/RHF/helper_cchbar.py

```diff
@@ -0,0 +1,254 @@
+    def build_Hvv(self):
+        self.Hvv = self.get_F('vv').copy()
+        self.Hvv -= ndot('ma,me->ae', self.t1, self.get_F('ov'))
+        self.Hvv += ndot('mf,amef->ae', self.t1, self.Lvovv)
+        self.Hvv -= ndot('mnfa,mnfe->ae', self.build_tau(), self.Loovv)
+        return self.Hvv
+
+    # <mn|Hbar|ij> = W_mnij + 0.25 * tau_ijef <mn||ef> = <mn||ij> + P(ij) t_je <mn||ie> + 0.5 * tau_ijef <mn||ef> 
+    def build_Hoooo(self):
+        self.Hoooo = self.get_MO('oooo').copy()
+        self.Hoooo += ndot('je,mnie->mnij', self.t1, self.get_MO('ooov'))
+        self.Hoooo += ndot('ie,mnej->mnij', self.t1, self.get_MO('oovo'))
+        self.Hoooo += ndot('ijef,mnef->mnij', self.build_tau(), self.get_MO('oovv'))
+        return self.Hoooo
+
+    # <ab|Hbar|ef> = W_abef + 0.25 * tau_mnab <mn||ef> = <ab||ef> - P(ab) t_mb <am||ef> + 0.5 * tau_mnab <mn||ef> 
+    def build_Hvvvv(self):
+        self.Hvvvv = self.get_MO('vvvv').copy()
+        self.Hvvvv -= ndot('mb,amef->abef', self.t1, self.get_MO('vovv'))
+        self.Hvvvv -= ndot('ma,bmfe->abef', self.t1, self.get_MO('vovv'))
+        self.Hvvvv += ndot('mnab,mnef->abef', self.build_tau(), self.get_MO('oovv'))
+        return self.Hvvvv
+
+    # <am|Hbar|ef> = <am||ef> - t_na <nm||ef> 
+    def build_Hvovv(self):
+        self.Hvovv = self.get_MO('vovv').copy()
+        self.Hvovv -= ndot('na,nmef->amef', self.t1, self.get_MO('oovv'))
+        return self.Hvovv
+
+    # <mn|Hbar|ie> = <mn||ie> + t_if <mn||fe> 
+    def build_Hooov(self):
+        self.Hooov = self.get_MO('ooov').copy()
+        self.Hooov += ndot('if,mnfe->mnie', self.t1, self.get_MO('oovv'))
+        return self.Hooov
+
+    # <mb|Hbar|ej> = W_mbej - 0.5 * t_jnfb <mn||ef> = <mb||ej> + t_jf <mb||ef> - t_nb <mn||ej> - (t_jnfb + t_jf t_nb) <nm||fe>   
+    def build_Hovvo(self):
+        self.Hovvo = self.get_MO('ovvo').copy()
+        self.Hovvo += ndot('jf,mbef->mbej', self.t1, self.get_MO('ovvv'))
+        self.Hovvo -= ndot('nb,mnej->mbej', self.t1, self.get_MO('oovo'))
+        self.Hovvo -= ndot('jnfb,nmfe->mbej', self.build_tau(), self.get_MO('oovv'))
+        self.Hovvo += ndot('jnbf,nmfe->mbej', self.t2, self.Loovv)
+        return self.Hovvo
+
+    # <mb|Hbar|je> = - <mb|Hbar|ej> = <mb||je> + t_jf <bm||ef> - t_nb <mn||je> - (t_jnfb + t_jf t_nb) <nm||ef>   
+    def build_Hovov(self):
+        self.Hovov = self.get_MO('ovov').copy()
+        self.Hovov += ndot('jf,bmef->mbje', self.t1, self.get_MO('vovv'))
+        self.Hovov -= ndot('nb,mnje->mbje', self.t1, self.get_MO('ooov'))
+        self.Hovov -= ndot('jnfb,nmef->mbje', self.build_tau(), self.get_MO('oovv'))
+        return self.Hovov
+
+    # <ab|Hbar|ei> = <ab||ei> - F_me t_miab + t_if Wabef + 0.5 * tau_mnab <mn||ei> - P(ab) t_miaf <mb||ef> 
+    #                - P(ab) t_ma {<mb||ei> - t_nibf <mn||ef>} 
+    def build_Hvvvo(self):
+        # <ab||ei>
+        self.Hvvvo =  self.get_MO('vvvo').copy()
```

Comment:
Can you break this function up a bit? Perhaps a space between each comment and code block?

Reply:
Done!